### PR TITLE
intel-oneapi-dnn-2024.1.0: add DNN package version

### PR DESCRIPTION
**Pull request by prstrnn**

### CODE REVIEW

File: var/spack/repos/builtin/packages/intel-oneapi-dnn/package.py [hunk 1]
```diff
@@ -26,6 +26,12 @@ class IntelOneapiDnn(IntelOneApiLibraryPackage):
         "https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onednn.html"
     )
 
+    version(
+        "2024.1.1",
+        url="'https://registrationcenter-download.intel.com/akdlm//IRC_NAS/5f6d82fa-2580-4bb1-83bb-cce7a52d1d34/l_onednn_p_2024.1.1.16_offline.sh",
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/intel-oneapi-dnn/package.py]:
```suggestion
        url="https://registrationcenter-download.intel.com/akdlm//IRC_NAS/5f6d82fa-2580-4bb1-83bb-cce7a52d1d34/l_onednn_p_2024.1.1.16_offline.sh",
```

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/intel-oneapi-dnn/package.py]: I can only confirm the sha256 with the above fix.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Minor typo but version sha256 confirmed once fixed.

**Review by rscohn2 — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Comment by tldahlgren** [comment]
Looks like a problem with some of the mac CI runners.

@spackbot run pipeline

**Pull request merged**